src/garage/np/_functions.py

```diff
@@ -1,7 +1,67 @@
 """Utility functions for NumPy-based Reinforcement learning algorithms."""
 import numpy as np
 
+from garage._dtypes import TrajectoryBatch
 from garage.misc import tensor_utils
+from garage.sampler.utils import rollout
+
+
+def process_paths(paths):
+    """Return processed sample data based on the collected paths.
+
+    Args:
+        paths (list[dict]): A list of collected paths.
+
+    Returns:
+        dict: Processed sample data, with keys
+            * undiscounted_returns (list[float])
+            * success_history (list[float])
+            * complete (list[bool])
+
+    """
+    success_history = [
+        path['success_count'] / path['running_length'] for path in paths
+    ]
+    undiscounted_returns = [path['undiscounted_return'] for path in paths]
+
+    # check if the last path is complete
+    complete = [path['dones'][-1] for path in paths]
+
+    samples_data = dict(undiscounted_returns=undiscounted_returns,
+                        success_history=success_history,
+                        complete=complete)
+
+    return samples_data
+
+
+def obtain_evaluation_samples(policy, env, max_path_length=1000,
+                              num_trajs=100):
+    """Sample the policy for 10 trajectories and return average values.
```

Comment:
I think It shouldn't be 10 trajectories but `num_trajs`.

Reply:
Thanks, fixed. I agree generally speaking, though this refactor didn't involve much logic changing, mainly moving things out of `OffPolicyRLAlgorithm`.